stdlib/stomp/src/main/java/org/ballerinalang/stdlib/stomp/endpoint/tcp/server/InitListener.java

```diff
@@ -0,0 +1,114 @@
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.stdlib.stomp.DefaultStompClient;
+import org.ballerinalang.stdlib.stomp.StompUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.ballerinalang.stdlib.stomp.StompConstants.STOMP_PACKAGE;
+import static org.ballerinalang.stdlib.stomp.StompConstants.CONFIG_FIELD_HOST;
+import static org.ballerinalang.stdlib.stomp.StompConstants.CONFIG_FIELD_PORT;
+import static org.ballerinalang.stdlib.stomp.StompConstants.CONFIG_FIELD_LOGIN;
+import static org.ballerinalang.stdlib.stomp.StompConstants.CONFIG_FIELD_PASSCODE;
+import static org.ballerinalang.stdlib.stomp.StompConstants.CONFIG_FIELD_CLIENT_OBJ;
+
+/**
+ * Initialize the server stomp endpoint.
+ *
+ * @since 0.990.2
+ */
+@BallerinaFunction(
+        orgName = "ballerina",
+        packageName = "stomp",
+        functionName = "initListener",
+        receiver = @Receiver(type = TypeKind.OBJECT, structType = "Listener", structPackage = STOMP_PACKAGE),
+        isPublic = true
+)
+public class InitListener extends BlockingNativeCallableUnit {
+    private static final Logger log = LoggerFactory.getLogger(InitListener.class);
+    private static String userLogin;
+    private static String userPasscode;
+    private static String hostName;
+    private static Integer stompPort;
+
+    @Override
+    public void execute(Context context) {
+        try {
+            BMap<String, BValue> connection = (BMap<String, BValue>) context.getRefArgument(0);
+
+            this.validateURI(context);
+            String connectionURI = "tcp://" + this.userLogin + ":" + this.userPasscode + "@" + this.hostName + ":" + this.stompPort;
+
+            DefaultStompClient client = new DefaultStompClient(new URI(connectionURI));
+            connection.addNativeData(CONFIG_FIELD_CLIENT_OBJ, client);
+
+            context.setReturnValues();
+        } catch (URISyntaxException e) {
+            context.setReturnValues(StompUtils.getError(context, e));
+        }
+    }
+
+    public void validateURI(Context context) {
+        BMap<String, BValue> endpointConfig = (BMap<String, BValue>) context.getRefArgument(1);
+
+            // TODO try to get rid the port by introducing only host with ex: tcp//localhost:61613
+            BString login = (BString) endpointConfig.get(CONFIG_FIELD_LOGIN);
+            if (login != null) {
+                this.userLogin = String.valueOf(login);
+            } else {
+                log.error("Login field is null");
+            }
+            //this.userLogin = String.valueOf(endpointConfig.get(CONFIG_FIELD_LOGIN));
+
+            BString passcode = (BString) endpointConfig.get(CONFIG_FIELD_PASSCODE);
+            if (passcode != null) {
+                this.userPasscode = String.valueOf(passcode);
+            } else {
+                log.error("Passcode field is null");
```

Comment:
IMHO not having a passcode is not an error. So we should have a default value. I think it can never be null because already we have empty string in bal code.